app/src/main/java/ch/epfl/sdp/peakar/points/ComputePOIPoints.java

```diff
@@ -0,0 +1,135 @@
+ * is visible or not. If no points are available or if they are not computed yet, they will be null
+ */
+public class ComputePOIPoints {
+    public static List<POIPoint> POIPoints;
+    public static Map<POIPoint, Boolean> labeledPOIPoints;
+    public static Map<POIPoint, Boolean> highestPOIPoints;
+    public static UserPoint userPoint;
+
+    private static final int HALF_MARKER_SIZE = 3;
+
+    /**
+     * Constructor of ComputePOIPoints, updates userPoint and gets the POIs for the userPoint
+     * @param context Context of activity
+     */
+    public ComputePOIPoints(Context context){
+        POIPoints = new ArrayList<>();
+        userPoint = UserPoint.getInstance(context);
+        userPoint.update();
+        getPOIs(userPoint);
+    }
+
+    /**
+     * Gets the POIs for the userPoint
+     * @param userPoint location of the user
+     */
+    private static void getPOIs(UserPoint userPoint){
+        new GeonamesHandler(userPoint){
+            @Override
+            public void onResponseReceived(ArrayList<POI> result) {
+                if(result!=null){
+                    for(POI poi : result){
+                        POIPoint poiPoint = new POIPoint(poi);
+                        poiPoint.setHorizontalBearing(userPoint);
+                        poiPoint.setVerticalBearing(userPoint);
+                        POIPoints.add(poiPoint);
+                    }
+                    getLabeledPOIs(userPoint);
+                }
+            }
+        }.execute();
+    }
+
+    /**
+     * Gets the labeled POIs and filters them
+     * @param userPoint userPoint for which the labeled POIs are computed
+     */
+    private static void getLabeledPOIs(UserPoint userPoint){
+        new DownloadTopographyTask(){
+            @Override
+            public void onResponseReceived(Pair<int[][], Double> topography) {
+                super.onResponseReceived(topography);
+                LineOfSight lineOfSight = new LineOfSight(topography, userPoint);
+                labeledPOIPoints = lineOfSight.getVisiblePointsLabeled(POIPoints);
+                //Filter highest mountains
+                highestPOIPoints = filterHighestPOIs(labeledPOIPoints);
+            }
+        }.execute(userPoint);
+    }
+
+    /**
+     * Filters the labeled POIPoints and takes only the highest s.t. there are no other markers
+     * within 6° horizontally. POIPoints in the line of sight are prioritised. If a POIPoint is in
+     * the line of sight but has a smaller altitude then another POIPoint that is close but that is
+     * not in the line of sight, then only the POIPoint that is in the line of sight will be added
+     * @param labeledPOIPoints labeled POIPoints that get sorted
+     * @return Map<POIPoint, Boolean> of the highest filtered POIs
+     */
+     static Map<POIPoint, Boolean> filterHighestPOIs(Map<POIPoint, Boolean> labeledPOIPoints) {
```

Comment:
Why is it only important to check the horizontal degrees and not vertical. 
Won't it become issues if you are near mountains. and see two mountains one smaller near you and one larger behind the smaller one. So you actually see both mountains. Will only the label for the one behind show?

Reply:
I do agree, although I think this needs some real-world tests as it is difficult to predict the various situations that might occur.